src/main/java/edu/kis/powp/jobs2d/visitor/ExceededCanvasVisitor.java

```diff
@@ -0,0 +1,38 @@
+package edu.kis.powp.jobs2d.visitor;
+
+import edu.kis.powp.jobs2d.command.*;
+
+import java.awt.*;
+import java.awt.geom.Rectangle2D;
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.List;
+
+public class ExceededCanvasVisitor implements ICommandVisitor{
+
+    private Shape area;
+    public boolean exceeds = false;
+
+    public ExceededCanvasVisitor(Shape area){
+        this.area = area;
+    }
+
+    @Override
+    public void visit(OperateToCommand command) {
+        exceeds = !area.contains(command.getPosX(), command.getPosY());
+    }
+
+    @Override
+    public void visit(SetPositionCommand command) {
+        exceeds = !area.contains(command.getPosX(), command.getPosY());
+    }
+
+    @Override
+    public void visit(ICompoundCommand command) {
+        Iterator<DriverCommand> iterator = command.iterator();
+        while(!exceeds && iterator.hasNext()) {
```

Comment:
You should set exceeds to true, before starting the loop. Otherwise, reusing the same instance of visitor might be error-prone.